### PR TITLE
Improved README

### PR DESCRIPTION
**Pull request by JamiesWhiteShirt**
It's about time we get a proper README. Suggestions welcome.

### CODE REVIEW

**Comment by Player3324** [comment]
It should explicitly point to checkstyle.xml in the root, which applies to all sub projects of Fabric API and needs to be configured in the IDE's Checkstyle plugin/support. A build should be also attempted since that ultimately verifies everything.

I'll provide further contribution guidelines at a later time as checkstyle can't cover everything.

**Comment by liach** [comment]
Consider targeting 1.14 as it is the current default branch.

**Comment by modmuss50** [comment]
I think the contribution guidelines should ideally be in a different document. Most people when looking at this will want to know how to download and use it. Possibly find out a bit more about what it does. 

Then there can be a section briefly (no more than what this has) about how to contribute that eventually leads to a more in-depth doc.

**Comment by natanfudge** [comment]
Needs to point to the fabric wiki as the first thing.

**Comment by i509VCB** [comment]
I feel adding these additional details is important to the readme, of course more reviews and suggestions would be nice.

File: README.md [hunk 1]
```diff
@@ -1,7 +1,25 @@
...
+
 ## Getting Started
 
 To setup a Fabric development environment, check out the [example fabric mod](https://github.com/FabricMC/fabric-example-mod) and follow the instructions there.
+
+## Contributing
+
+[Checkstyle](https://checkstyle.sourceforge.io/) is used to enforce code style. Make sure to install the necessary tools in your development environment to not be held up in review by failing code style checks.
+
+For more information on contributing to Fabric, read [The Fabric Feature Procedure](https://fabricmc.net/wiki/tutorial:feature_procedure).
```
Comment (i509VCB) [inline README.md:24]: Honestly I feel going full out with a `CONTRIBUTING.MD` file would suit this much better.

**Review by i509VCB — COMMENTED** [review]
Maybe a list of modules and a quick description of what they do would be nice?

**Comment by i509VCB** [comment]
Superseded by #1109

**Pull request closed without merging**